services/venice-router/src/main/java/com/linkedin/venice/router/RouterServer.java

```diff
@@ -985,8 +978,8 @@ public VeniceRouterConfig getConfig() {
   }
 
   public void setReadRequestThrottling(boolean throttle) {
-    RouterThrottler throttler = throttle ? readRequestThrottler : noopRequestThrottler;
-    scatterGatherMode.initReadRequestThrottler(throttler);
+    readRequestThrottler.setIsNoopThrottlerEnabled(throttle);
+    scatterGatherMode.initReadRequestThrottler(readRequestThrottler);
```

Comment:
There is no need to call `scatterGatherMode.initReadRequestThrottler` here since the throttler instance never changes, right?

Reply:
Actually `initReadRequestThrottler` is never called anywhere else, so we could move the init call to `startService` and remove from here, or keep it as is.

Reply:
probably need to add some test here,  it should be `readRequestThrottler.setIsNoopThrottlerEnabled(!throttle);` but surprised no test caught it.